### PR TITLE
Service Registration: Templates

### PR DESCRIPTION
**Pull request by kmh287**
This pull request distinguishes installing/registering the templates service and creating it.Please see #7839 for more context.

/to @choumx

### CODE REVIEW

File: src/service/template-impl.js [hunk 1]
```diff
@@ -326,14 +333,7 @@ export class Templates {
  * @package
  */
 export function registerExtendedTemplate(win, type, templateClass) {
-  return installTemplatesService(win).registerTemplate_(type, templateClass);
+  installTemplatesService(win);
```
Comment (choumx) [inline src/service/template-impl.js]: Might not be needed since it's installed in runtime.js.

Comment (kmh287) [inline src/service/template-impl.js]: done.

**Comment by kmh287** [comment]
@willchou Let me know if you have any further comments

File: test/functional/test-template.js [hunk 2]
```diff
@@ -19,7 +19,8 @@ import {
   installTemplatesService,
   registerExtendedTemplate,
 } from '../../src/service/template-impl';
-
+import {templatesFor} from '../../src/services';
+import {resetServiceForTesting} from '../../src/service';
```
Comment (choumx) [inline test/functional/test-template.js]: Nit: Alphabetize.

Comment (kmh287) [inline test/functional/test-template.js]: done

**Pull request merged**